version: '3'
services:

  walletapi:
    image: waltid/ssikit
    ports:
      - "19010:7002"
    restart: no
    entrypoint: /app/bin/waltid-ssikit serve -b 0.0.0.0
    volumes:
      - './waltIdApiData/walletapidata:/app/data'
    networks:
      - network

  issuerapi:
    image: waltid/ssikit
    ports:
      - "19011:7001"
    restart: no
    entrypoint: /app/bin/waltid-ssikit serve -b 0.0.0.0
    volumes:
      - './waltIdApiData/issuerapidata:/app/data'
    networks:
      - network

  verifierapi:
    image: waltid/ssikit
    ports:
      - "19012:7003"
    restart: no
    entrypoint: /app/bin/waltid-ssikit serve -b 0.0.0.0
    volumes:
      - './waltIdApiData/verifierapidata:/app/data'
    networks:
      - network

  dbissuerapp:
    image: postgres
    restart: no
    environment:
      POSTGRES_PASSWORD: admin
      POSTGRES_USER: admin
      POSTGRES_DB: issuerapp
    networks:
      network:
        ipv4_address: 10.5.0.5
    ports:
      - 5432:5432

  adminer:
    image: adminer
    restart: always
    ports:
      - 8080:8080
    networks:
      - network

networks:
  network:
    driver: bridge
    ipam:
      config:
        - subnet: 10.5.0.0/16
          gateway: 10.5.0.1
volumes:
  db_data: {}